### ISSUE TITLE
Clicking the breakpoint label triggers the checkbox focus

### ISSUE THREAD

**Issue by digitarald** [firefox-devtools/debugger#8026]
STR:
- Add a breakpoint
- Slowly lick the breakpoint label (not the checkbox)

AR: Checkbox gets a focused state, looking like it will get toggled
ER: Checkbox doesn't change when clicking, as the label should only jump to the location

![screenflow](https://user-images.githubusercontent.com/8599/53374123-487c3080-390c-11e9-81c0-4a8f212fcc95.gif)

**Comment by fvsch** [comment]
Personally I'm unhappy with the interactions for this whole row.

We have three actions:
1. disable breakpoint (checkbox, always visible)
2. remove breakpoint (cross icon, visible on row hover only)
3. navigate to source (invisible, no specific signifier or visual feedback)

Having one interaction perceptible on hover only and a second one not perceptible at all until you click is not a great situation, to begin with. On top of that, those two interactions are not available using keyboard navigation.

We also have a full-row hover style which is very ambiguous, given how we offer 3 different click interactions in that row.

Some options:
- Remove the hover effect.
- The checkbox could have no visible label (click the checkbox to toggle); we will need an accessible label then (title + aria-label?).
- I would keep the `line:column` info always visible, and link _that_. Linking a location is consistent with what we have in Inspector and Console. A `text-decoration: underline` on hover + a tooltip would help make that interaction clear (see locations in Inspector > Rules, and log locations in Console).
- The code snippet could have zero interactions. Or it could have the same function as the `line:column`, if we show similar interactive feedback (or if we join both together). It's probably worth exploring a few options in mockups.

**Comment by jasonLaster** [comment]
@fvsch I think these are great points for a larger issue that we should re-evaluate a Breakpoint's UX #8042. 

I'd like to fix this small UI nit here because #8042 could take longer than this upcoming release :)

**Comment by fvsch** [comment]
@jasonLaster Good point. I think for this issue we should just strive to remove the <label> in `src/components/SecondaryPanes/Breakpoints/Breakpoint.js`.

To keep things roughly accessible (though the current label is not a great accessible description IMO, but that can be reworked in #8042), we should try to keep the same label association though. Schematically, going from:

```html
<input type="checkbox" id="foo"/>
<label for="foo">...</label>
```

to:

```html
<input type="checkbox" id="foo" aria-labelledby="foo-label"/>
<span id="foo-label">...</span>
```
